src/Sarif/Autogenerated/ExternalProperties.cs

```diff
@@ -38,6 +38,7 @@ public virtual SarifNodeKind SarifNodeKind
         /// The URI of the JSON schema corresponding to the version of the external property file format.
         /// </summary>
         [DataMember(Name = "schema", IsRequired = false, EmitDefaultValue = false)]
+        [JsonConverter(typeof(Microsoft.CodeAnalysis.Sarif.Readers.UriConverter))]
         public virtual Uri Schema { get; set; }
```

Comment:
>Uri [](http://example.com/codeflow?start=23&length=3)

There's a 1x1 mapping between this type, URI and the JsonConverter attribute. That could have been the way we handled this in the codegen hints + jschema (i.e., provide a way to emit a specific attribute for every instance of a property with a specific return type). Just an idea, can you confirm this feature does not already exist?

Reply:
The UriConverter class is from SDK not JSchama so we can not directly always generate this attribute when Uri type in JSchema.

We could have support hint that base on type, not just name, then we can have all this declared once. I checked the hint is only based on name.

An other way is we could have still use  base on name but support partial wild card e.g. `*.*.Uri*` then we can have it declared declared once, except this kind of special one `Schema` that does no have Uri in the name. 
But I checked we only support * at the first part only: `*.Uri`.

```CSharp
private static string MakeHintDictionaryKey(string typeName, string propertyName)
        {
            return typeName + "." + propertyName.ToPascalCase();
        }
```

Let me know I can create a work item and work on it.